src/shader/shader.rs

```diff
@@ -101,8 +100,8 @@ impl GraphicsShader {
 impl Drop for GraphicsShader {
     fn drop(&mut self) {
         unsafe {
-            self.device.vk().destroy_shader_module(self.vertex_shader, ALLOCATION_CALLBACKS);
-            self.device.vk().destroy_shader_module(self.fragment_shader, ALLOCATION_CALLBACKS);
```

Comment:
what happened with `ALLOCATION_CALLBACKS`?

Reply:
Right completely forgot that. I removed it.

Reasoning is as follows: 
- Doing custom allocators for vulkan objects is highly complex and more likely than not will just degrade performance unless we setup some very powerful system.
- Properly supporting allocation callbacks would require more than just some static handle to our code. So our current implementation wouldnt have been useful even if we wanted to implement it.

Reply:
Also from the spec is this:
> Since most memory allocations are off the critical path, this is not meant as a performance feature. Rather, this can be useful for certain embedded systems, for debugging purposes (e.g. putting a guard page after all host allocations), or for memory allocation logging.
So considering this i don't it makes sense for use to implement custom allocators for vulkan